server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -308,6 +312,25 @@ Runnable getGlobalCheckpointSyncer() {
     private final Store remoteStore;
     private final TranslogFactory translogFactory;
 
+    public boolean isBlockInternalCheckPointRefresh() {
+        return blockInternalCheckPointRefresh;
```

Comment:
Is it possible to create this boolean tracker variable elsewhere? May be segmentReplicationTargetService through an index event update? just thinking aloud.

Reply:
but i get this should be set in a place such that both the source and target could access it through a standard access pattern.